morebits.js

```diff
@@ -1738,6 +1740,7 @@ Morebits.wiki.api.prototype = {
 	statusText: null, // result received from the API, normally "success" or "error"
 	errorCode: null, // short text error code, if any, as documented in the MediaWiki API
 	errorText: null, // full error description, if any
+	tokenRetry: false,
```

Comment:
Looks like you renamed this to `badtokenRetry` below (or vice-versa)

Reply:
Indeed. So what messed me up here was the later realisation that all actions don't use a CSRF token. If we want to be fussy about it, we could do something like [what I do in my bot framework](https://github.com/siddharthvp/mwn/blob/master/src/bot.js#L525) (querying the type of token needed as well) but IMO not really necessary. 

Easier to pretend that everything is CSRF-based (badtoken error on anything which isn't will be failing as before).

Reply:
Indeed, that's roughly where I ended up thinking.  It's few and far between atm, so nbd.